src/client.rs

```diff
@@ -294,6 +310,8 @@ impl Client {
     where
         T: Into<Option<u64>>,
     {
+        self.check_sync_state()?;
```

Comment:
So the backoff is only for `sync()`? I would have expected this check only in the `get()` method, i.e. when we are actually hitting the server. If we simply retrieve a value from a local cache, the backoff should be irrelevant.

Reply:
We only hit the server in `sync()` actually. The `get()` reads from cache by default, and calls `sync()` on empty databases. Consumers should call `sync()` explicitly if they use a persistent storage (ie. not dummy).

So, yes the idea is to prevent any client that was told to backoff to contact the server again.

Reply:
Ah, sorry, got that mixed up.